src/InteractiveWindow/Editor/InteractiveWindow.UIThreadOnly.cs

```diff
@@ -2468,71 +2603,241 @@ private void CutOrDelete(IEnumerable<SnapshotSpan> projectionSpans, bool isCut)
+                        }
+                    }
+
+                    if (dataHasBoxCutCopyTag)
+                    {
+                        if (selection.IsEmpty && IsCaretOnBlankEditableLine())
+                        {
+                            InsertText(code);
+                        }
+                        else
+                        {
+                            VirtualSnapshotPoint unusedStart, unusedEnd;
+                            EditorOperations.InsertTextAsBox(code, out unusedStart, out unusedEnd);
+                        }
+                    }
+                    else
+                    {
+                        InsertText(code);
+                    }
+
+                    // don't record undo histroy for standard-input
+                    if (!ReadingStandardInput)
+                    {
+                        transaction.Complete();
+                    }
+                    return true;
+                }
+            }
+            
+            /// <summary>Implements <see cref="IInteractiveWindowOperations.Cut"/>.</summary>
+            public void Cut()
+            {
+                using (var transaction = UndoHistory_TestOnly.CreateTransaction(InteractiveWindowResources.Cut))
+                {
+                    if (TextView.Selection.IsEmpty)
+                    {
+                        CopyCurrentLine();
+                        if (!DeleteCurrentLine())
+                        {
+                            return;
+                        }
+                        MoveCaretToClosestEditableBuffer();
+                    }
+                    else
+                    {
+                        var selection = TextView.Selection;
+                        CopySelection();
+                        if (!DeleteSelection())
+                        {
+                            return;
+                        }
+
+                        if (selection.Mode == TextSelectionMode.Box)
+                        {
+                            ReduceBoxSelectionToEditableBox(isDelete: true);
+                        }
+                        else
+                        {
+                            selection.Clear();
+                            MoveCaretToClosestEditableBuffer();
+                        }
+                        TextView.Caret.EnsureVisible();
+                    }
+
+                    // don't record undo histroy for standard-input
+                    if (!ReadingStandardInput)
+                    {
+                        transaction.Complete();
+                    }
                 }
             }
 
             /// <summary>Implements <see cref="IInteractiveWindowOperations2.Copy"/>.</summary>
             public void Copy()
             {
-                CopySelection();
+                if (TextView.Selection.IsEmpty)
+                {
+                    CopyCurrentLine();
```

Comment:
`Selection.IsEmpty` would be `false` if it's a box selection (even if the box selection is actually empty), so it will call `CopySelection` instead.